src/Storages/StorageDistributed.cpp

```diff
@@ -546,6 +549,14 @@ void StorageDistributed::read(
 
     ClusterProxy::executeQuery(query_plan, select_stream_factory, log,
         modified_query_ast, context, query_info);
+
+    if (!query_plan.isInitialized())
```

Comment:
Note that this place conflicts with #21550

Reply:
Are there any cases where query_plan is not initialized other than empty shard set? If no, I'd suggest moving the fix to the top of this function body.

Reply:
And in this case it may also conflicts with #21526 , but will see